ycmd/completers/rust/rust_completer.py

```diff
@@ -82,7 +74,7 @@ def _GetResponse( self, handler, request_data = {} ):
     were found.
     """
     self._logger.info('RustCompleter._GetResponse')
-    target = urlparse.urljoin( self._racerd_host, handler )
+    target = 'http://localhost:' + str(self._racerd_port) + str(handler)
```

Comment:
We are aware of the race condition with `utils.GetUnusedLocalhostPort` but we use it with:

- YouCompleteMe spawning ycmd
- Spawing OmnisharpServer
- Spawning JediHTTP (in the future :stuck_out_tongue_closed_eyes: )
- Spawning Tern.js (there is a PR for Tern.js support)

The problem with reading the initial output from racerd is that for doing that you should wait for the server to start printing stuff on stdout instead of telling the server to start and move on. Plus it makes the code much simpler.